specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,264 @@
+# Background
+
+We are exposing an event that will fire when an attempt to launch a registered protocol is made. The host will be given the option to cancel the launch, handle the popup dialog to hide the dialog, as well as revoke permissions that have been previously given to the origin launching the protocol. 
+
+# Description
+
+This event will fire before the registered protocol launch occurs. Currently a popup dialog is displayed in which the user can click `Open` or `Cancel`. If the request is made from a trustworthy origin a checkmark box will be displayed that will allow the user to always allow this registered protocol from this origin. The `NavigationStarting`, `NavigationCompleted`, `SourceChanged`, `ContentLoading`, and `HistoryChanged` events will not fire when a request is made to launch a registered protocol. 
```

Comment:
What does it mean for an origin to be trustworthy or not? Can the end dev control that? Should the end dev be able to obtain this info so they can do something similar in their own replacement UI?

Reply:
It calls IsOriginPotentiallyTrustworthy on the origin, making sure it's not opaque, if the scheme is https or wss, if it's localhost scheme, a filescheme, if the scheme is considered authenticated by the user agent, or if the origin has been explicitly allowlisted. 
It's mostly origin dependent but the end dev can run their app in an environment where the origin is explicitly configured to be allowlisted. Given the teams scenario ask I don't think it's necessary for the end dev to obtain this info/ haven't seen asks for this on GH.

Reply:
We need to explain in our docs what 'trustworthy origin' means. Is it documented somewhere else? If so we can link to that. If not, and its not something that the end dev has control over then maybe we can say something like 'origins are deemed trustworthy by heuristics built into the web platform' or something.

Reply:
The phrase trustworthy origin is also in the reference docs in the API section below and would also need to be defined there for our ref docs